### PR TITLE
Feature/teams rework

### PR DESCRIPTION
**Pull request by AlexandraMash**
The following features were implemented:
#162 
#155 
#145 
#143

### CODE REVIEW

File: src/lib/services/achievements_service.dart [hunk 1]
```diff
@@ -128,10 +137,10 @@ class AchievementsService {
...
-        .getUserAchievements(_authService.currentUser.id)
-        .then((list) =>
-            list.map((a) => AchievementModel.fromAchievement(a)).toList());
-  }
+  // Future<List<AchievementModel>> getMyAchievements() {
+  //   return _achievementsDatabaseService
+  //       .getUserAchievements(_authService.currentUser.id)
+  //       .then((list) =>
+  //           list.map((a) => AchievementModel.fromAchievement(a)).toList());
+  // }
```
Comment (AlexandraMash) [inline src/lib/services/achievements_service.dart:145]: Comments should be removed

File: src/lib/viewmodels/team_member_picker_viewmodel.dart [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      return 0; // not a member
+    } else if (_teamMembers[user.id].accessLevel == UserAccessLevel.member) {
+      return 1; // member
+    } else {
+      return 2; // admin
+    }
+  }
+
+  void saveChanges(BuildContext context) async {
+    if (_teamMembers.isEmpty) {
```
Comment (AlexandraMash) [inline src/lib/viewmodels/team_member_picker_viewmodel.dart:68]: Maybe it's also necessary to check if there is at least one member with access level "admin"

File: src/lib/viewmodels/team_member_picker_viewmodel.dart [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    notifyListeners();
+  }
+
+  int getUserState(UserModel user) {
+    if (!_teamMembers.containsKey(user.id)) {
+      return 0; // not a member
+    } else if (_teamMembers[user.id].accessLevel == UserAccessLevel.member) {
+      return 1; // member
+    } else {
+      return 2; // admin
```
Comment (wcoder) [inline src/lib/viewmodels/team_member_picker_viewmodel.dart:63]: seems like local enum

File: src/lib/services/achievements_service.dart (hunk 1 above)
Comment (SavikPavel) [reply src/lib/services/achievements_service.dart]: Fixed

File: src/lib/viewmodels/team_member_picker_viewmodel.dart (hunk 2 above)
Comment (SavikPavel) [reply src/lib/viewmodels/team_member_picker_viewmodel.dart]: Fixed

File: src/lib/services/achievements_service.dart [hunk 4]
```diff
@@ -22,8 +22,17 @@ class AchievementsService {
...
-    return _achievementsDatabaseService.getTeamsAchievements().then((list) =>
-        list.map((a) => AchievementModel.fromAchievement(a)).toList());
+    var teamsAchievements =
+        await _achievementsDatabaseService.getTeamsAchievements();
+
+    var myAchievements = await _achievementsDatabaseService
+        .getUserAchievements(_authService.currentUser.id);
+
+    var achievementsSet = Set.from(teamsAchievements);
+    achievementsSet.addAll(myAchievements);
```
[the remark this answers is not in the document]
Comment (SavikPavel) [reply src/lib/services/achievements_service.dart]: Reworked

File: src/lib/services/database/teams_database_service.dart [hunk 5]
```diff
@@ -16,6 +16,16 @@ class TeamsDatabaseService {
             value.documents.map((d) => Team.fromJson(d.data, d.documentID)));
   }
 
+  Future<Iterable<Team>> getPublicTeams() {
+    return _database
+        .collection(_teamsCollection)
+        .where("access_level", isEqualTo: 0)
```
[the remark this answers is not in the document]
Comment (SavikPavel) [reply src/lib/services/database/teams_database_service.dart]: Fixed

File: src/lib/viewmodels/team_member_picker_viewmodel.dart (hunk 3 above)
Comment (SavikPavel) [reply src/lib/viewmodels/team_member_picker_viewmodel.dart]: Fixed

File: src/assets/icons/crown.svg [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	c4.819,4.091,11.68,4.746,17.22,1.55l123.72-71.53l11.97-41.57L350.661,345.081z"/>
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
```
[the remark this answers is not in the document]
Comment (SavikPavel) [reply src/assets/icons/crown.svg]: Fixed

**Pull request merged**